### PR TITLE
Introduce lightning connect

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds the missing documentation for the Lightning App configure_api / configure_commands. 

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

**Comment by tchaton** [comment]
Note for the reviewer: The Configure API docs will change once https://github.com/Lightning-AI/lightning/pull/14183 is merged

**Review by Felonious-Spellfire — APPROVED** [review]
Made updates directly to the files.

File: docs/source-app/glossary/command_lines/command_lines.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+**Audience:** Users looking to create a command line interface (CLI) for their application.
+
+----
+
+**************
+What is a CLI?
+**************
+
+A CLI is a text-based user interface (UI) used to run programs, manage computer files and interact with the computer. CLIs are also called command-line user interfaces, console user interfaces and character user interfaces.
```
Comment (awaelchli) [inline docs/source-app/glossary/command_lines/command_lines.rst:15]: We need a citation or write it in our own words.

File: docs/source-app/workflows/build_command_line_interface/cli.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    lightning run app app.py
+
+The following appears the terminal:
+
+.. code-block::
+
+    Your Lightning App is starting. This won't take long.
+    INFO: Your app has started. View it in your browser: http://127.0.0.1:7501/view
+    []
```
Comment (awaelchli) [inline docs/source-app/workflows/build_command_line_interface/cli.rst:35]:
```suggestion
```

File: docs/source-app/workflows/build_command_line_interface/index.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+############################
+Command-line Interface (CLI)
```
Comment (awaelchli) [inline docs/source-app/workflows/build_command_line_interface/index.rst:2]: why does this page exist twice?

**Review by awaelchli — APPROVED** [review]

File: docs/source-app/glossary/command_lines/command_lines.rst (hunk 1 above)
Comment (tchaton) [reply docs/source-app/glossary/command_lines/command_lines.rst:15]: Done !

File: docs/source-app/workflows/build_command_line_interface/cli.rst (hunk 2 above)
Comment (tchaton) [reply docs/source-app/workflows/build_command_line_interface/cli.rst:35]: This is added on purpose. On the first execution, the self.names is empty.

File: docs/source-app/workflows/build_command_line_interface/index.rst (hunk 3 above)
Comment (tchaton) [reply docs/source-app/workflows/build_command_line_interface/index.rst:2]: Because the bottom of the page is different.

**Review by manskx — APPROVED** [review]

**Pull request merged**